src/player.cpp

```diff
@@ -8977,7 +8977,8 @@ item player::reduce_charges( int position, long quantity )
         debugmsg( "invalid item position %d for reduce_charges", position );
         return ret_null;
     }
-    if( it.reduce_charges( quantity ) ) {
+    it.mod_charges( -quantity );
```

Comment:
Is there a guarantee that `reduce_charges` won't be invoked with `quantity > charges`?